layer/layer.go

```diff
@@ -199,11 +199,11 @@ func createChainIDFromParent(parent ChainID, dgsts ...DiffID) ChainID {
 		return parent
 	}
 	if parent == "" {
-		return createChainIDFromParent(ChainID(dgsts[0]), dgsts[1:]...)
+		return createChainIDFromParent(ChainID(dgsts[0]), dgsts[1:]...) //nolint:gosec // FIXME(thaJeztah): ignore G602: slice index out of range, which is a false positive
 	}
 	// H = "H(n-1) SHA256(n)"
-	dgst := digest.FromBytes([]byte(string(parent) + " " + string(dgsts[0])))
-	return createChainIDFromParent(ChainID(dgst), dgsts[1:]...)
+	dgst := digest.FromBytes([]byte(string(parent) + " " + string(dgsts[0]))) //nolint:gosec // FIXME(thaJeztah): ignore G602: slice index out of range, which is a false positive
+	return createChainIDFromParent(ChainID(dgst), dgsts[1:]...)               //nolint:gosec  // FIXME(thaJeztah): ignore G602: slice index out of range, which is a false positive
 }
```

Comment:
I tried to create a minimal reproducer for this one, but for some reason GolangCI only picked up this case, but not when trying to write a more minimal one. I'll give it another try to see if I can report this.